File: Seminar/src/main/java/org/sopt/Seminar/member/domain/SOPT.java
package org.sopt.Seminar.member.domain;

import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import static jakarta.persistence.EnumType.*;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class SOPT {
	private int generation;

	@Enumerated(value = STRING)
	private Part part;
}

File: kopring/week2/kotlin-semina/src/main/kotlin/org/sopt/kotlinsemina/domain/member/model/SOPT.kt
package org.sopt.kotlinsemina.domain.member.model

import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class SOPT(
    val generation: Int,

    @Enumerated(EnumType.STRING)
    val part: Part,
) {
}